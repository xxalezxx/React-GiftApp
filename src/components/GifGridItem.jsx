import React from "react"
import PropTypes from "prop-types"

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="animate__animated animate__bounce animate__delay-2s">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

export default GifGridItem
