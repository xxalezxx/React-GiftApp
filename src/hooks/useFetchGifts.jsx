import { useState, useEffect } from "react"

import { getGif } from "../helpers/getGifts"

export const useFetchGifts = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  })

  useEffect(() => {
    getGif(category).then((imgs) => {
      setState({
        data: imgs,
        loading: false,
      })
    })
  }, [category])

  return state
}
