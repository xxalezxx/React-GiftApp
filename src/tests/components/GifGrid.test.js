import { shallow } from "enzyme"
import React from "react"
import "@testing-library/jest-dom"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifts } from "../../hooks/useFetchGifts"

jest.mock("../../hooks/useFetchGifts")

describe("Testing of GifGrid.js", () => {
  const category = "batman"

  test("Should match with the snapshot", () => {
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test("Should show items when it loads images", () => {
    const gifs = [
      {
        id: "abc",
        url: "asdasdasdasdadassd",
        title: "nada",
      },
      {
        id: "123",
        url: "asdasdasdasdadassd",
        title: "nada",
      },
    ]
    useFetchGifts.mockReturnValue({
      data: gifs,
      loading: false,
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find("p").exists()).toBe(false)
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
  })
})
