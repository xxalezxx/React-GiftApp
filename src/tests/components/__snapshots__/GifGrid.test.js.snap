// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of GifGrid.js Should match with the snapshot 1`] = `
<Fragment>
  <h3>
    batman
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing of GifGrid.js Should show items when it loads images 1`] = `
<Fragment>
  <h3>
    batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="nada"
      url="asdasdasdasdadassd"
    />
    <GifGridItem
      id="123"
      key="123"
      title="nada"
      url="asdasdasdasdadassd"
    />
  </div>
</Fragment>
`;
